src/app/drive/components/DriveExplorer/DriveExplorer.tsx

```diff
@@ -301,9 +302,11 @@ const DriveExplorer = (props: DriveExplorerProps): JSX.Element => {
     (item: DriveItemData | AdvancedSharedItem) => {
       if (item.isFolder) {
         dispatch(storageThunks.goToFolderThunk({ name: item.name, id: item.id }));
+        navigationService.pushFolder(item.uuid);
       } else {
         dispatch(uiActions.setIsFileViewerOpen(true));
         dispatch(uiActions.setFileViewerItem(item as DriveItemData));
+        navigationService.pushFile(item.uuid);
```

Comment:
It is necessary the previous dispatch in order to trigger the FileViewer?

Reply:
Yep, because the FileViewer is placed in App.tsx, so we can use the FileViewerWrapper component for SharedView, no matter if the user is logged in or not.

Reply:
I think we are not talking about the same thing. 
In Drive at least, with the new navigation you don't need to use the `dispatch` of those `uiActions` used here, just add the uuid of the file or folder with the navigationService and it opens automatically. 
Check if it works as expected without the previous dispatches:
```
dispatch(uiActions.setIsFileViewerOpen(true));
dispatch(uiActions.setFileViewerItem(item as DriveItemData)); 
```

Reply:
Ah okay. I'll take a look.

Reply:
Hey! I've just checked if it works without the dispatch and you was right 🙌🏽